### ISSUE TITLE
Question about the loss function.

### ISSUE THREAD

**Issue by dawuchen** [wentianli/knowledge_distillation_caffe#5]
Hello @wentianli .
Thanks for your code for knowledge distillation.
I have one question about your loss function equation of "loss -= soft_label[pos] * (log(std::max(prob_data[pos], Dtype(FLT_MIN)))-log(std::max(soft_label[pos], Dtype(FLT_MIN))));
", you count loss by letting student 's softmax output minus teacher's output ,why do you do like this?What if sometimes student 's output greater than teacher's output and sometimes on the contrary ?As I see it ,they may neutralize.

**Comment by wentianli** [comment]
```
for (int c = 0; c < bottom[0]->shape(softmax_axis_); ++c) { # sum for every class
    pos = i * dim + c * inner_num_ + j;
    loss -= soft_label[pos] * (log(prob_data[pos])-log(soft_label[pos]));
    }
```
You are right that the loss for some classes is positive, and for others negative, but when we sum them up (for every class), the result is always non-negative.  [KL divergence](https://en.wikipedia.org/wiki/Kullback–Leibler_divergence) is always non-negative, please refer to section _Properties_ in wikipedia.

**Comment by dawuchen** [comment]
Thanks,now I know it.
